app/controllers/admin/routes.py

```diff
@@ -157,6 +159,54 @@ def rsvpLogDisplay(eventId):
     else:
         abort(403)
 
+@admin_bp.route('/event/<eventId>/renew', methods=['POST'])
+def renewEvent(eventId):
+    try: 
+        formData=request.form
+        try:
+            assert formData['timeStart'] < formData['timeEnd']
+        except AssertionError:
+            flash("End time must be after start time", 'warning')
+            return redirect(url_for('admin.eventDisplay', eventId = eventId))
+        
+        try:
+            if formData.get('dateEnd'):
+                assert formData['dateStart'] < formData['dateEnd']
+        except AssertionError:
+            flash("End date must be after start date", 'warning')
+            return redirect(url_for('admin.eventDisplay', eventId = eventId))
+
+
+        priorEvent = model_to_dict(Event.get_by_id(eventId))
+        newEventDict = priorEvent.copy()
+        newEventDict.pop('id')
+        newEventDict.update({
+                    'program': int(priorEvent['program']['id']),
+                    'term': int(priorEvent['term']['id']),
+                    'timeStart': formData['timeStart'],
+                    'timeEnd': formData['timeEnd'],
+                    'location': formData['location'],
+                    'startDate': f'{formData["startDate"][-4:]}-{formData["startDate"][0:-5]}',
+                    'endDate': f'{formData["endDate"][-4:]}-{formData["endDate"][0:-5]}',
+                    'isRecurring': bool(priorEvent['recurringId'])
+                    })
+        newEvent, message = attemptSaveEvent(newEventDict, renewedEvent = True)
+        if message:
+            flash(message, "danger")
+            return redirect(url_for('admin.eventDisplay', eventId = eventId))
+
+        copyRsvp(priorEvent['id'], newEvent[0].id)
+        createAdminLog(f"Renewed {priorEvent['name']} as <a href='event/{newEvent[0].id}/view'>{newEvent[0].name}</a>.")
+        flash("Event successfully renewed.", "success")
+        return redirect(url_for('admin.eventDisplay', eventId = newEvent[0].id))
+
+
+    except Exception as e:
+        print("Error while trying to renew event:", e)
```

Comment:
remove print statement

Reply:
To my understanding, this is the proper way to do this and the same way it is done in other places to show exceptions in flask.